lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -427,11 +424,11 @@ def base_url
       end
 
       def avs_result(response)
-        response['source'] && response['source']['avs_check'] ? AVSResult.new(code: response['source']['avs_check']) : nil
+        response.respond_to?(:dig) && response.dig('source', 'avs_check') ? AVSResult.new(code: response['source']['avs_check']) : nil
```

Comment:
Checking the `respond_to` since the response is an integer if the action is unstore. Question for reviewers: do you feel more comfortable with this handling or based on action type?

Reply:
I think this implementation is fine. A comment wouldn't hurt, but that's up to you